lib/synapse_pay_rest/http_client.rb

```diff
@@ -52,49 +52,75 @@ def update_headers(user_id: nil, oauth_key: nil, fingerprint: nil, client_id: ni
 
     def post(path, payload)
       url = base_url + path
-      response = begin
-        RestClient.post(url,
-          payload.to_json,
-          @headers)
-      rescue Exception => e
-        return JSON.parse(e.response)
-      end
-      return JSON.parse(response)
+      response = with_error_handling { RestClient.post(url, payload.to_json, @headers) }
+      JSON.parse(response)
     end
 
     def patch(path, payload)
       url = base_url + path
-      response = begin
-        RestClient.patch(url,
-          payload.to_json,
-          @headers)
-      rescue Exception => e
-        puts url
-        return JSON.parse(e.response)
-      end
-      return JSON.parse(response)
+      response = with_error_handling { RestClient.patch(url, payload.to_json, @headers) }
+      JSON.parse(response)
     end
 
     def get(path)
       url = base_url + path
-      response = begin
-        RestClient.get(url,
-          @headers)
-      rescue Exception => e
-        return JSON.parse(e.response)
-      end
-      return JSON.parse(response)
+      response = with_error_handling { RestClient.get(url, @headers) }
+      JSON.parse(response)
     end
 
     def delete(path)
       url = base_url + path
-      response = begin
-        RestClient.delete(url,
-          @headers)
-      rescue Exception => e
-        return JSON.parse(e.response)
+      response = with_error_handling { RestClient.delete(url, @headers) }
+      JSON.parse(response)
+    end
+
+    private
+
+    def with_error_handling
+      yield
+    rescue => e
+      # By the way, this is a really bad idea.
+      # See: https://www.relishapp.com/womply/ruby-style-guide/docs/exceptions
+      # The exceptions should be enumerated. Not all exceptions are going
+      # to be parsable by JSON. The only one that should be captured are the
+      # are the HTTP Client responses.
+      case e.response.code
+      when 400
+        return e.response
+      when 401
+        return e.response
+      when 409
+        return e.response
+      when 500
+        return e.response
```

Comment:
I think there may be a bug in the v12 code.  When we were looking over the errors, did we find that the 500 errors were returning html?  Maybe it returns both?